src/tools/illink/src/linker/Linker.Steps/RootAssemblyInputStep.cs

```diff
@@ -86,6 +86,9 @@ protected override void Process ()
 
 				// No metadata trimming
 				Context.MetadataTrimming = MetadataTrimming.None;
+
+				// Library mode builds against System.Runtime ref assembly
+				Context.SystemModuleName = "System.Runtime";
```

Comment:
We should know what the core assembly is (it's the one that defines System.Object). Why is this extra configuration needed? I think I didn't fully understand https://github.com/dotnet/runtime/pull/104060#discussion_r1655659027.